disnake/types/activity.py

```diff
@@ -82,37 +84,31 @@ class ActivityEmoji(_ActivityEmojiOptional):
     name: str
 
 
-class ActivityButton(TypedDict):
-    label: str
-    url: str
-
-
 class _SendableActivityOptional(TypedDict, total=False):
     url: Optional[str]
 
 
-ActivityType = Literal[0, 1, 2, 4, 5]
+ActivityType = Literal[0, 1, 2, 3, 4, 5]
 
 
 class SendableActivity(_SendableActivityOptional):
     name: str
     type: ActivityType
 
 
-class _BaseActivity(SendableActivity):
-    created_at: int
-
-
-class Activity(_BaseActivity, total=False):
-    state: Optional[str]
-    details: Optional[str]
+class Activity(SendableActivity, total=False):
+    created_at: int  # required according to docs, but we treat it as optional for simplicity
```

Comment:
Why does this make it simpler?

Reply:
`created_at` isn't required when a user instantiates one of the activity types, but some `to_dict` methods return full `types.activity.Activity` dicts which results in typing issues if `created_at` were required.

We could remove some of the `to_dict` methods and only keep the ones that actually return a `SendableActivity`, since the others aren't useful to bots. However, the one use case I can see for them is serializing and storing activities somewhere, so keeping all of them might make sense as well.